internal/transport/http2_server.go

```diff
@@ -1104,6 +1104,9 @@ func (t *http2Server) WriteStatus(s *Stream, st *status.Status) error {
 	}
 	// Send a RST_STREAM after the trailers if the client has not already half-closed.
 	rst := s.getState() == streamActive
+	if st.Code() == codes.DeadlineExceeded {
```

Comment:
can you write a test for this or modify an exsting test to verify?

Reply:
How did you verify if this is the only change we need? Did you test we are actually getting RST_STREAM?

Reply:
@purnesh42H  Please find the screen once putting the sleep time 40sec, deadline exceed we are getting if we are remove the check "context cancel" respectively ![image](https://github.com/user-attachments/assets/bf7b0285-2936-43b2-b3ee-602e0a155f82)

Reply:
you need to write a test to verify where client is receiving RST_STREAM when deadline exceeded. Check existing tests where RST_STREAM is sent

Reply:
sure @purnesh42H  I will do same